aip/client-libraries/4222.md

```diff
@@ -154,6 +169,46 @@ If none of the routing parameters matched their respective fields, the routing h
 Much like URL parameters, if there is more than one key-value pair to be sent, the `&`
 character is used as the separator.
 
+### `path_template` syntax
+
+As seen in the above examples, the `path_template` can use a variety of symbols
+that are interpreted by code generators during conversion to regular expressions
+or non-regular expression matcher implementations. The `path_template` consists
+of segments delimited by the segment delimiter. The syntax for `path_template`
+is as follows:
+
+- The only acceptable segment delimiter is `/`.
+  - The last symbol in a `path_template` **may** be a delimiter - it will be 
+    ignored.
+- A segment **must** be of one of the following types:
+  - `*`: A single-segment wildcard. Corresponds to 1 or more non-`/` symbols.
+    The regex describing it is `[^\/]+`.
+    - A Single-segment wildcard typically represents a resource ID.
+  - `**`: A multi-segment wildcard. Corresponds to 0 or more symbols or segments
+    (so it can be empty). Consumes segment delimiters and does not match unless
+    the delimiter has trailing content. The regex describing it is `([:\/].*)?`.
+    - A multi-segment wildcard **must** only appear as the final segment.
+  - `LITERAL`: A literal segment. A literal segment can contain any
+    alphanumeric symbol.
+    - A literal segment **must not** contain a symbol reserved in this syntax.
+    - Literal segments typically represent a resource collection ID or base
+      path.
+  - `{}`: A variable segment. This matches part of the path as specified by its
+    template.
+    - A variable segment can be either of the following:
+      - `{key}`, where `key` is the name to be used in the key-value pair of the
+        header
+      - `{key=template}`, where the `template` is the segment(s) (expressed in
+        this `path_template` syntax) to extract as the value paired with `key`
+    - A variable segment of just `{key}` defaults to a template of `*` which
+      matches everything non-empty. 
```

Comment:
"1 or more non-`/` symbols" like in the "*" segment description is probably a better way to put it.

Reply:
For sure, done.